File: src/main/java/no/jstien/bikeapi/tsdb/read/StationHistory.java
package no.jstien.bikeapi.tsdb.read;

public class StationHistory {
    private int stationId;
    private TimeSerie freeBikes;
    private TimeSerie freeLocks;

    public StationHistory(int stationId) {
        this.stationId = stationId;
        this.freeBikes = new TimeSerie();
        this.freeBikes = new TimeSerie();
    }

    public StationHistory(int stationId, TimeSerie freeBikes, TimeSerie freeLocks) {
        this.stationId = stationId;
        this.freeBikes = freeBikes;
        this.freeLocks = freeLocks;
    }

    void setFreeBikes(TimeSerie freeBikes) {
        this.freeBikes = freeBikes;
    }

    void setFreeLocks(TimeSerie freeLocks) {
        this.freeLocks = freeLocks;
    }


    public int getStationId() {
        return stationId;
    }

    public TimeSerie getFreeBikes() {
        return freeBikes;
    }

    public TimeSerie getFreeLocks() {
        return freeLocks;
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/read/StationHistory.kt
package no.jstien.bikeapi.tsdb.read

class StationHistory {
    var stationId: Int = 0
        private set
    var freeBikes: TimeSerie? = null
        internal set
    var freeLocks: TimeSerie? = null
        internal set

    constructor(stationId: Int) {
        this.stationId = stationId
        this.freeBikes = TimeSerie()
        this.freeBikes = TimeSerie()
    }

    constructor(stationId: Int, freeBikes: TimeSerie, freeLocks: TimeSerie) {
        this.stationId = stationId
        this.freeBikes = freeBikes
        this.freeLocks = freeLocks
    }
}
